scipy/stats/tests/test_sensitivity_analysis.py

```diff
@@ -164,31 +169,61 @@ def jansen_sobol(f_A, f_B, f_AB):
             """Jansen for S and Sobol' for St.
 
             From Saltelli2010, table 2 formulations (c) and (e)."""
-            s_, n = f_A.shape
-            f_AB = f_AB.reshape((-1, s_, n))
-
-            var = np.var(np.concatenate([f_A, f_B], axis=1), axis=1)
+            var = np.var([f_A, f_B], axis=(0, -1))
 
             s = (var - 0.5*np.mean((f_B - f_AB)**2, axis=-1)) / var
             st = np.mean(f_A*(f_A - f_AB), axis=-1) / var
 
-            return s.reshape(s_, -1), st.reshape(s_, -1)
+            return s.T, st.T
 
         rng = np.random.default_rng(28631265345463262246170309650372465332)
         res = sobol_indices(
             func=f_ishigami, n=4096,
-            dists=[
-                stats.uniform(loc=-np.pi, scale=2*np.pi),
-                stats.uniform(loc=-np.pi, scale=2*np.pi),
-                stats.uniform(loc=-np.pi, scale=2*np.pi)
-            ],
+            dists=self.dists,
             method=jansen_sobol,
             random_state=rng
         )
 
         assert_allclose(res.first_order, ishigami_ref_indices[0], atol=1e-2)
         assert_allclose(res.total_order, ishigami_ref_indices[1], atol=1e-2)
 
+    def test_normalization(self, ishigami_ref_indices):
+        rng = np.random.default_rng(28631265345463262246170309650372465332)
+        res = sobol_indices(
+            func=lambda x: f_ishigami(x) + 1000, n=4096,
+            dists=self.dists,
+            random_state=rng
+        )
+
+        assert_allclose(res.first_order, ishigami_ref_indices[0], atol=1e-2)
+        assert_allclose(res.total_order, ishigami_ref_indices[1], atol=1e-2)
+
+    def test_constant_function(self, ishigami_ref_indices):
+
+        def f_ishigami_vec_const(x):
+            """Output of shape (3, n)."""
+            res = f_ishigami(x)
+            return np.concatenate([res, res * 0 + 10, res]).reshape(3, -1)
```

Comment:
```suggestion
        def f_ishigami_vec_const(x):
            """Output of shape (3, n)."""
            res = f_ishigami(x)
            return res, res * 0 + 10, res
```
This should do the job. It seems like it would at least be the right shape. I don't think we should require the user to return an array; it just needs to have the right shape.